### PR TITLE
Create changelogs for 3.14.1.0

### PR DESCRIPTION
**Pull request by Kleidukos**
Forward-port of https://github.com/haskell/cabal/pull/10576

### CODE REVIEW

**Review by geekosaur — APPROVED** [review]
Minor nit ("which" is fine informally but I think changelogs should be a little more formal).

File: release-notes/Cabal-3.14.1.0.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+## Cabal and Cabal-syntax 3.14.1.0 changelog
+
+### Significant changes
+
+- Fix build ways for modules in executables [#10418](https://github.com/haskell/cabal/issues/10418) [#10419](https://github.com/haskell/cabal/pull/10419)
+
+  - Modules belonging to executables were being built in too many ways. For instance, if you
+  had configured to build profiled library files then your executable modules would also
+  be built profiled. Which was a regression in behaviour since `Cabal-3.12`.
```
[the remark this answers is not in the document]
Comment (Kleidukos) [reply release-notes/Cabal-3.14.1.0.md:9]: Sorry I think I'm going to pass on this suggestion. This should have probably been raised during the PR that introduced the changelog entry, or even during https://github.com/haskell/cabal/pull/10576, but I'm not comfortable tweaking the text of changelogs between `master` and their base branch.

**Comment by Mikolaj** [comment]
This is still borked: https://github.com/haskell/cabal/pull/10576#issuecomment-2492517870

**Review by Mikolaj — APPROVED** [review]
The two typos I've described are gone on this branch (what about 3.14 branch?). Thank you for including the 3.14.0.0 changelogs. If they were included in a separate section, it would be easy to tell what changed between 3.14.0.0 and 3.14.1.0, but it's probably a rare use case and the way it's set up makes it easier to read for a person that switches from 3.12.* to 3.14.1.0 directly, which is the common case, so I don't object. I haven't proof-read everything carefully, so another pair of eyes would be welcome. Maybe @ffaf1 finds a moment?

File: release-notes/Cabal-3.14.1.0.md [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+  NAME[:COMPONENT_NAME]=CID
+  ```
+
+- Add support for building profiled dynamic way [#4816](https://github.com/haskell/cabal/issues/4816) [#9900](https://github.com/haskell/cabal/pull/9900)
+
+  Add support for profiled dynamic way
+
+  New options for `cabal.project` and `./Setup` interface:
+
+  * `profiling-shared`: Enable building profiling dynamic way
```
Comment (ffaf1) [inline release-notes/Cabal-3.14.1.0.md:74]:
```suggestion
  * `profiling-shared`: Enable building profiling dynamic way.
```

File: release-notes/Cabal-3.14.1.0.md [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+    -> Maybe (SymbolicPath CWD (Dir dir1))
+
+    -> [SymbolicPath dir1 (Dir dir2)]
+
+    -> RelativePath dir2 File
+
+    -> IO (SymbolicPath dir1 File)
+  ```
+
+  See Note [Symbolic paths] in `Distribution.Utils.Path` for further information
```
Comment (ffaf1) [inline release-notes/Cabal-3.14.1.0.md:116]:
```suggestion
  See `Note [Symbolic paths]` in `Distribution.Utils.Path` for further information
```

**Review by ffaf1 — APPROVED** [review]

**Pull request merged**